pkg/manager/manager.go

```diff
@@ -335,11 +335,21 @@ func New(config *rest.Config, options Options) (Manager, error) {
 	}
 
 	// Create the resource lock to enable leader election)
-	leaderConfig := options.LeaderElectionConfig
-	if leaderConfig == nil {
+	var leaderConfig *rest.Config
+	var leaderRecorderProvider *intrec.Provider
+
+	if options.LeaderElectionConfig == nil {
 		leaderConfig = rest.CopyConfig(config)
+		leaderRecorderProvider = recorderProvider
+	} else {
+		leaderConfig = rest.CopyConfig(options.LeaderElectionConfig)
```

Comment:
The config is a pointer to a *rest.Config. We set some fields down the line for User-Agent and similiar which means we manipulate the value behind the pointer, so it appears in the original kubeconfig, i.E.:

```
leCfg := GetConfig()
mgr.New(cfg, manager.Options{LeaderElectionConfig: leCfg}
// If we don't do the CopyConfig, the User-Agent in the leCfg got changed
```